### ISSUE TITLE
PyInstaller build possibly incomplete

### ISSUE THREAD

**Issue by BeanBagKing** [log2timeline/plaso#4658]
**Describe the problem:**

**Bottom line up front**: Windows build of psteal.exe seems to run fine, but produces results that, most significantly, have missing information ("N/A") in the Source and Source Long column, and <WARNING DEFAULT FORMATTER> in the message column.

I'm trying to build, and document for the future the method of building, Windows bundles. I completely understand that log2timeline no longer provides these. I am hoping that, given ~4 years of progress on PyInstaller (which I understand was the main issue), this might be re-evaluated. Failing that, I'm hoping for some help on getting the documented build process more fully fleshed out. Failing that as well, I hope that those in control of these issues will at least let this one remain open for a while to allow for community discussion.

I believe I have the build process working. I appear to have a version of the log2timeline tools that passes the -vast- majority of run_tests.py prior to building (and those it doesn't pass appear to be cosmetic issues), has all of the dependencies, builds without error, and runs without error. That said, the output (resulting csv from psteal.exe) has issues. I'm hoping someone more familiar with the project might be able to hint at what those are.

**To Reproduce:**

The version of Plaso you used: 20230413

The operating system you are running Plaso on: Windows 11 Pro 22H2

Steps to reproduce the behavior including command line and arguments and output: Please see: https://nullsec.us/log2timeline-windows-build/
I could paste it all here, but frankly, that would screw up the majority of the formatting at this point and be an absolute mess to read. If anyone wants more detailed output from any of the steps, please let me know and I can re-run them and provide that here.

TL;DR is:

* Install git/c++/python 3.11/pyinstaller
* git clone plaso / l2tdevtools
* install plaso pre-reqs
* make_release.ps1
* plaso.exe runs, but the output is... off
    * **most significantly**, source and source_long are "N/A" and every message is prepended with "<WARNING DEFAULT FORMATTED>" - if this sounds familiar, I'll take any hints.

Please provide the source data you used when you experienced the problem: Windows 11 VMDK

The method you used to install Plaso: Custom build of plaso.exe (see reproduction steps)

**Expected behavior:**

Output consistent with other versions of plaso/l2t

**--troubles output**
```
C:\plaso-20230413-amd64>log2timeline.exe --troubles
2023-05-17 23:58:31,189 [INFO] (MainProcess) PID:20388 <data_location> Determined data location: C:\plaso-20230413-amd64\data
Using Python version 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]

Path: C:\plaso-20230413-amd64\log2timeline.py

plaso - log2timeline version 20230413

Checking availability and versions of dependencies.
[OK]            acstore version: 20230325
[OK]            artifacts version: 20230413
[OK]            bencode
[OK]            certifi version: 2023.05.07
[OK]            cryptography version: 40.0.2
[OK]            dateutil version: 2.8.2
[OK]            defusedxml version: 0.7.1
[OK]            dfdatetime version: 20230506
[OK]            dfvfs version: 20230507
[OK]            dfwinreg version: 20221218
[OK]            dtfabric version: 20230512
[OPTIONAL]      unable to determine version information for: flor
[OK]            future version: 0.18.3
[OK]            lz4 version: 4.3.2
[OK]            opensearchpy
[OK]            pefile version: 2023.2.7
[OK]            psutil version: 5.9.5
[OK]            pybde version: 20221031
[OK]            pycaes version: 20230406
[OK]            pycreg version: 20221022
[OK]            pyesedb version: 20230318
[OK]            pyevt version: 20221022
[OK]            pyevtx version: 20221101
[OK]            pyewf version: 20230212
[OK]            pyfsapfs version: 20221102
[OK]            pyfsext version: 20220829
[OK]            pyfsfat version: 20220925
[OK]            pyfshfs version: 20220831
[OK]            pyfsntfs version: 20230427
[OK]            pyfsxfs version: 20220829
[OK]            pyfvde version: 20220915
[OK]            pyfwnt version: 20220922
[OK]            pyfwsi version: 20230114
[OK]            pylnk version: 20230205
[OK]            pyluksde version: 20221103
[OK]            pymodi version: 20221023
[OK]            pymsiecf version: 20221024
[OK]            pyolecf version: 20221024
[OK]            pyparsing version: 3.0.9
[OK]            pyphdi version: 20221025
[OK]            pyqcow version: 20221124
[OK]            pyregf version: 20230319
[OK]            pyscca version: 20221027
[OK]            pysigscan version: 20230109
[OK]            pysmdev version: 20221028
[OK]            pysmraw version: 20230320
[OK]            pytsk3 version: 20221228
[OK]            pytz
[OK]            pyvhdi version: 20221124
[OK]            pyvmdk version: 20221124
[OK]            pyvsgpt version: 20221029
[OK]            pyvshadow version: 20221030
[OK]            pyvslvm version: 20221025
[OK]            redis version: 4.5.5
[OK]            requests version: 2.30.0
[OK]            six version: 1.16.0
[OK]            urllib3 version: 1.26.15
[OPTIONAL]      missing: xattr.
[OK]            xlsxwriter version: 3.1.0
[OK]            yaml version: 6.0
[OK]            yara version: 4.3.1
[OK]            zmq version: 25.0.2

Also see: https://plaso.readthedocs.io/en/latest/sources/user/Troubleshooting.html

C:\plaso-20230413-amd64>
```

**Comment by joachimmetz** [comment]
We don't support exe builds any  more. If files are missing from a pyinstaller build, check the build logs, change the build configuration if necessary

**Comment by joachimmetz** [comment]
We recommend using Docker instead.

**Comment by BeanBagKing** [comment]
Hi joachimmetz, I'm the author of the nullsec.us article!

The comment you mentioned regarding community involvement was what I was hoping to get here. If you don't want to support or help with this, that's fine. I realize log2timeline has no interest in publishing Windows builds anymore, but log2timeline does have instructions, binaries, and helper files across three different projects to allow the community to do so themselves. If I can get this working, I'd be happy to contribute back to those. I don't mind going at this alone, and already put quite a bit of time into it, but it's going to be exponentially more difficult if I can't even discuss the problems here that I'm seeing with other people interested in the log2timeline project(s).

Regarding the rest of your comments and the title change:

What makes you think this is a PyInstaller issue? This seems more like an undocumented dependency I'm missing to me, but I fully admit I'm not a developer. However, the build seems to produce valid executable that not only work without issue during runtime, but seem to be parsing data correctly from a disk image. Even if it's just a guess at this point, I'd like to know what, from your expertise, what makes you think so. It might give me a lead to follow. 

Can you tell me what you mean by "files missing from the pyinstaller build". All the prerequisites seem to be there, both in the run_tests.py that was checked before pyinstaller, and the --troubles output from the executable themselves. Maybe I'm misunderstanding what missing files you are talking about though. If this isn't Plaso dependencies, what other files (generically) do you think would be missing?

What build logs should I be looking at? I mean this both specifically (where would I find these, if you don't mind holding my hand a bit), and more generically? Do you mean the build logs from the original plaso build (e.g. python plaso\setup.py build), or the logs (if there are any?) produced by make_release.ps1 (or some sub-process resulting from that)?

What build configuration are you referencing? the make_release.ps1, or plaso\setup.py build, or some other configuration?

Regarding Docker, while I appreciate the recommendation, this isn't a solution for every use case. Autopsy would be a great example, as they bundle a (very old at this point) executable with the application. I'm not part of that project at all, just pointing out that there are use cases where Docker isn't an appropriate solution.

**Comment by joachimmetz** [comment]
> already put quite a bit of time into it

yes that is unfortunately the downside of PyInstaller.

> This seems more like an undocumented dependency I'm missing to me, but I fully admit I'm not a developer. 

you need to make sure PyInstaller packages all dependencies and data files, that is a common issue with making PyInstaller builds

> However, the build seems to produce valid executable that not only work without issue during runtime

Please read up how PyInstaller works, it is basically an archive of the Python and data files and dependencies, with a runtime loader

> but it's going to be exponentially more difficult if I can't even discuss the problems here that I'm seeing with other people interested in the log2timeline project(s).

This is an issue tracker not a discussion forum for unsupported features, there is a slack channel for discussion

>  I'm not part of that project at all, just pointing out that there are use cases where Docker isn't an appropriate solution.

My time is very limited, if you want to support such use cases you'll have to help out.
